File: banda/SergioDeIscar-banda-java/SergioDeIscar-banda-java/src/main/java/org/example/models/Percusionista.java
package org.example.models;

import org.example.enums.PercussionType;
import org.example.interfaces.IPercusionista;

public final class Percusionista extends Musician implements IPercusionista {
    private final Float salario = super.salario * 1.35f;
    private PercussionType percussionType;
    public Percusionista(String nombre, int experiencia, PercussionType percussionType) {
        super(nombre, experiencia);
        setPercussionType(percussionType);
    }

    @Override
    public PercussionType getPercussionType() {
        return percussionType;
    }

    @Override
    public void setPercussionType(PercussionType percussionType) {
        this.percussionType = percussionType;
    }

    @Override
    public float getSalario() {
        return salario;
    }

    @Override
    public String getNombre() {
        return this.nombre;
    }
}

File: banda/SergioDeIscar-banda-kotlin/SergioDeIscar-banda-kotlin/src/main/kotlin/models/Percusionista.kt
package bandamusical.models

import bandamusical.interfeces.IPercusionista

class Percusionista(nombre:String,
                    override val experiencia: Int,
                    override val percussion: PercussionType): Musician(nombre), IPercusionista {
    override val salario: Float = super.salario * 1.35f
}

enum class PercussionType{
    ALTURA_DETERMINADA,
    ALTURA_INDETERMINADA
}